Error 409 when executing IBM_IAM_USER_INVITE

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform CLI and Terraform IBM Provider Version

Terraform 1.12.2
Ibmcloud Provider 1.87.3

### Affected Resource(s)

* ibm_iam_user_invite

### Terraform Configuration Files

```hcl
resource "ibm_iam_user_invite" "access_group_user_management" {
  users = ["gaston.lagaffe@bnpparibas.com"]
}
```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

```
ibm_iam_user_invite.access_group_user_management: Creating...
2026-02-11T21:20:58.897+0100 [INFO]  Starting apply for ibm_iam_user_invite.access_group_user_management
2026-02-11T21:20:58.898+0100 [DEBUG] ibm_iam_user_invite.access_group_user_management: applying the planned Create change
2026-02-11T21:20:58.899+0100 [INFO]  provider.terraform-provider-ibm_v1.87.3: 2026/02/11 21:20:58 [DEBUG] setting computed for "invited_users" from ComputedKeys: timestamp="2026-02-11T21:20:58.899+0100"
2026-02-11T21:20:58.899+0100 [INFO]  provider.terraform-provider-ibm_v1.87.3: 2026/02/11 21:20:58 {[] [] [] <nil> []}: timestamp="2026-02-11T21:20:58.899+0100"
2026-02-11T21:20:58.900+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: [DEBUG] REQUEST: [2026-02-11T21:20:58+01:00] POST /v2/accounts/364cb0da7b7c42b4ba83efc6876cf496/users HTTP/1.1
2026-02-11T21:20:58.900+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Host: user-management.cloud.ibm.com
2026-02-11T21:20:58.900+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Accept: application/json
2026-02-11T21:20:58.900+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Accept-Language: en
2026-02-11T21:20:58.900+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Authorization: [PRIVATE DATA HIDDEN]
2026-02-11T21:20:58.900+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Content-Type: application/json
2026-02-11T21:20:58.900+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: User-Agent: Bluemix-go SDK 0.1 / darwin
2026-02-11T21:20:58.900+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: X-Original-User-Agent: terraform-provider-ibm/1.87.3
2026-02-11T21:20:58.900+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3
2026-02-11T21:20:58.900+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: {"users":[{"email":"gaston.lagaffe@bnpparibas.com","account_role":"MEMEBER"}]}
2026-02-11T21:21:01.891+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: [DEBUG] RESPONSE: [2026-02-11T21:21:01+01:00] Elapsed: 2989ms HTTP/1.1 202 Accepted
2026-02-11T21:21:01.891+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Content-Length: 118
2026-02-11T21:21:01.891+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Akamai-Grn: 0.e3d84b17.1770841259.48e319e
2026-02-11T21:21:01.891+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Cache-Control: max-age=0, no-cache, no-store
2026-02-11T21:21:01.891+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Connection: Keep-Alive
2026-02-11T21:21:01.891+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Content-Type: application/json; charset=utf-8
2026-02-11T21:21:01.891+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Date: Wed, 11 Feb 2026 20:21:01 GMT
2026-02-11T21:21:01.891+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Expires: Wed, 11 Feb 2026 20:21:01 GMT
2026-02-11T21:21:01.891+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Ibm-Cloud-Service-Name: user-management
2026-02-11T21:21:01.892+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Ibm-Cloud-Tenant: 9e999102faa34be087b435a04bf59d95,15fc9828678f4b2b9820c1331ba39570,0e8d3cd94877445fa9764becb08ea5a7,33a7eb04e7d547cd9489e90c99d476a5,985c667b4f89413c974f812500e94ce0,81ada35eddd8453696febe365ad85ee2,a25f4fa98485407bb7b44e647f8a5e15,db8f0cc7bde24bbda8a71c121868f65a,ac3e80fb25684f758aa059ef66c0b1fc,afba06f61a974bc4bd8c36560a3416da,4a45f260564541068d8e1c83f29a9623,a3c1da9c4e4a4cdaaa92d1edeb7f4868,6b29e5d81a8e4adda23ab13fd4e8fb03,ff8d762413d34824b12e2d3427b270a2
2026-02-11T21:21:01.892+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Pragma: no-cache
2026-02-11T21:21:01.892+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Server: istio-envoy
2026-02-11T21:21:01.892+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Set-Cookie: ak_bmsc=269A4372C2CAACE501FD5CE6EFABA3B2~000000000000000000000000000000~YAAQ49hLF9FArUKcAQAAIUddTh665SMTodzlJbUeP7N6aAIgXHoANPPAVCQsGWmauh+yPm+txLqarSg4GTMgDMtxrH5Gg9W5fT3YkKj0t0uXRS9KK4AwjofOtXzNnJffSsa8Neq9E5kaF2EqRixK56uBBwREYDmeQ9BgNdgiMyEkgjUbrjUZRsyiPpzdFgNKpmLhkPq6pZNLXIpqaJGDtIjms1MS3frbvb9Oximgsek6mnvDPWeC71wtGk0mm8AW4gN20/R4wE+q1r5+fGRVBPHdaPXQ6xWeMJt84MYi4hP+LoPsnZS+BvaALFGMGF3es3y+kxQKXKOK4xvh+ysCXaoOWkjypzMXiNKZY92Ay/Y=; Domain=.cloud.ibm.com; Path=/; Expires=Wed, 11 Feb 2026 22:20:59 GMT; Max-Age=7198; HttpOnly
2026-02-11T21:21:01.892+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Strict-Transport-Security: max-age=31536000; includeSubDomains
2026-02-11T21:21:01.892+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Transaction-Id: f1e310f8-1983-4db4-a5fc-50abdb63264d
2026-02-11T21:21:01.892+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: X-Correlation-Id: 61db7ac4-1921-4049-a638-ac5df9ddcbd3
2026-02-11T21:21:01.892+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: X-Envoy-Upstream-Service-Time: 1847
2026-02-11T21:21:01.892+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: X-Ratelimit-Limit: 10
2026-02-11T21:21:01.892+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: X-Ratelimit-Remaining: 9
2026-02-11T21:21:01.892+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: X-Ratelimit-Reset: 1770841261
2026-02-11T21:21:01.892+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: X-Request-Id: 15d6ebf3-68a7-45e0-b180-343dd1717b4a
2026-02-11T21:21:01.892+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3
2026-02-11T21:21:01.892+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: {"resources":[{"id":"36e816dea6c54fc6972ed5920a30b678","email":"gaston.lagaffe@bnpparibas.com","state":"PROCESSING"}]}
2026-02-11T21:21:01.892+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: [DEBUG] REQUEST: [2026-02-11T21:21:01+01:00] POST /v2/accounts/364cb0da7b7c42b4ba83efc6876cf496/users HTTP/1.1
2026-02-11T21:21:01.892+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Host: user-management.cloud.ibm.com
2026-02-11T21:21:01.892+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Accept: application/json
2026-02-11T21:21:01.892+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Accept-Language: en
2026-02-11T21:21:01.892+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Authorization: [PRIVATE DATA HIDDEN]
2026-02-11T21:21:01.892+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Content-Type: application/json
2026-02-11T21:21:01.892+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: User-Agent: Bluemix-go SDK 0.1 / darwin
2026-02-11T21:21:01.892+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: X-Original-User-Agent: terraform-provider-ibm/1.87.3
2026-02-11T21:21:01.892+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3
2026-02-11T21:21:01.893+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: {"users":[{"email":"gaston.lagaffe@bnpparibas.com","account_role":"MEMEBER"}]}
2026-02-11T21:21:03.324+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: [DEBUG] RESPONSE: [2026-02-11T21:21:03+01:00] Elapsed: 1431ms HTTP/1.1 409 Conflict
2026-02-11T21:21:03.324+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Content-Length: 267
2026-02-11T21:21:03.324+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Akamai-Grn: 0.e3d84b17.1770841261.48e3592
2026-02-11T21:21:03.324+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Cache-Control: max-age=0, no-cache, no-store
2026-02-11T21:21:03.324+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Connection: Keep-Alive
2026-02-11T21:21:03.324+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Content-Type: application/json; charset=utf-8
2026-02-11T21:21:03.324+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Date: Wed, 11 Feb 2026 20:21:03 GMT
2026-02-11T21:21:03.324+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Expires: Wed, 11 Feb 2026 20:21:03 GMT
2026-02-11T21:21:03.324+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Ibm-Cloud-Service-Name: user-management
2026-02-11T21:21:03.325+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Ibm-Cloud-Tenant: 9e999102faa34be087b435a04bf59d95,15fc9828678f4b2b9820c1331ba39570,0e8d3cd94877445fa9764becb08ea5a7,33a7eb04e7d547cd9489e90c99d476a5,985c667b4f89413c974f812500e94ce0,81ada35eddd8453696febe365ad85ee2,a25f4fa98485407bb7b44e647f8a5e15,db8f0cc7bde24bbda8a71c121868f65a,ac3e80fb25684f758aa059ef66c0b1fc,afba06f61a974bc4bd8c36560a3416da,4a45f260564541068d8e1c83f29a9623,a3c1da9c4e4a4cdaaa92d1edeb7f4868,6b29e5d81a8e4adda23ab13fd4e8fb03,ff8d762413d34824b12e2d3427b270a2
2026-02-11T21:21:03.325+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Pragma: no-cache
2026-02-11T21:21:03.325+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Server: istio-envoy
2026-02-11T21:21:03.325+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Set-Cookie: ak_bmsc=17D0A94DD284AC862E01A5A6AEF310DF~000000000000000000000000000000~YAAQ49hLFwJBrUKcAQAAikxdTh764RPvWueeVc/cwBKTeIG/p80jKKdpUaQ9BTBTt/CqxvcFMVEIyaVNvwdUZwk/f+RLrlddhXzlpzDHUckVsUgEvHjjwU8DRs/Nus/AZemh1EYMoVtt/JjbCCLbbNM0ZRfwNke/XVdCHh3ENNRhJuue5eMZshmRt7PYQCl6YMCzugnNW39AL4sO5ALVDq0p3dyZ6Cb8hpEnsGo6/6n8Cjk54X+OJUNuUWEO1olwMM5+7sCEY9oFKzuJ72pc42T24TsP1KqPadGoZU0cG97i3Gv/NzDGdPwQPUxVo4LJX+3kfJFPWsXkrNsCPu0nyZULtQ0Nv3X4PrQ5zhQdXQQ=; Domain=.cloud.ibm.com; Path=/; Expires=Wed, 11 Feb 2026 22:21:01 GMT; Max-Age=7198; HttpOnly
2026-02-11T21:21:03.325+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Strict-Transport-Security: max-age=31536000; includeSubDomains
2026-02-11T21:21:03.325+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: Transaction-Id: 10778924-a33c-4684-b720-04d63a9867e2
2026-02-11T21:21:03.325+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: X-Correlation-Id: 87c76ae7-c9c6-47f6-b26c-45d9efd9e3e3
2026-02-11T21:21:03.325+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: X-Envoy-Upstream-Service-Time: 1284
2026-02-11T21:21:03.325+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: X-Ratelimit-Limit: 10
2026-02-11T21:21:03.325+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: X-Ratelimit-Remaining: 9
2026-02-11T21:21:03.325+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: X-Ratelimit-Reset: 1770841262
2026-02-11T21:21:03.325+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: X-Request-Id: a59d71f3-04cf-4b37-a73e-e1927bd1b69c
2026-02-11T21:21:03.325+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3
2026-02-11T21:21:03.325+0100 [DEBUG] provider.terraform-provider-ibm_v1.87.3: {"trace":"10778924-a33c-4684-b720-04d63a9867e2","errors":[{"code":"Conflict","message":"Invite request blocked : gaston.lagaffe@bnpparibas.com","more_info":"the following user(s) are already in processing state, please try again after one minute"}],"status_code":409}
2026-02-11T21:21:03.325+0100 [INFO]  provider.terraform-provider-ibm_v1.87.3: 2026/02/11 21:21:03 [DEBUG] ---
id: terraform-fa2a7c87
summary: 'Request failed with status code: 409, ServerErrorResponse: {"trace":"10778924-a33c-4684-b720-04d63a9867e2","errors":[{"code":"Conflict","message":"Invite
  request blocked : gaston.lagaffe@bnpparibas.com","more_info":"the following user(s)
  are already in processing state, please try again after one minute"}],"status_code":409}'
severity: error
resource: ibm_iam_user_invite
operation: create
component:
  name: github.com/IBM-Cloud/terraform-provider-ibm
  version: 1.87.3
---: timestamp="2026-02-11T21:21:03.324+0100"
2026-02-11T21:21:03.325+0100 [ERROR] provider.terraform-provider-ibm_v1.87.3: Response contains error diagnostic: @module=sdk.proto
  diagnostic_detail=
  | ---
  | id: terraform-fa2a7c87
  | summary: 'Request failed with status code: 409, ServerErrorResponse: {"trace":"10778924-a33c-4684-b720-04d63a9867e2","errors":[{"code":"Conflict","message":"Invite
  |   request blocked : gaston.lagaffe@bnpparibas.com","more_info":"the following user(s)
  |   are already in processing state, please try again after one minute"}],"status_code":409}'
  | severity: error
  | resource: ibm_iam_user_invite
  | operation: create
  | component:
  |   name: github.com/IBM-Cloud/terraform-provider-ibm
  |   version: 1.87.3
  | ---
   diagnostic_severity=ERROR diagnostic_summary="Request failed with status code: 409, ServerErrorResponse: {\"trace\":\"10778924-a33c-4684-b720-04d63a9867e2\",\"errors\":[{\"code\":\"Conflict\",\"message\":\"Invite request blocked : gaston.lagaffe@bnpparibas.com\",\"more_info\":\"the following user(s) are already in processing state, please try again after one minute\"}],\"status_code\":409}" tf_proto_version=5.10 tf_resource_type=ibm_iam_user_invite tf_provider_addr=provider tf_req_id=4268adc6-fa0b-85d0-3aab-68ec19188619 tf_rpc=ApplyResourceChange @caller=github.com/hashicorp/terraform-plugin-go@v0.29.0/tfprotov5/internal/diag/diagnostics.go:58 timestamp="2026-02-11T21:21:03.325+0100"
2026-02-11T21:21:03.335+0100 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2026-02-11T21:21:03.335+0100 [ERROR] vertex "ibm_iam_user_invite.access_group_user_management" error: Request failed with status code: 409, ServerErrorResponse: {"trace":"10778924-a33c-4684-b720-04d63a9867e2","errors":[{"code":"Conflict","message":"Invite request blocked : gaston.lagaffe@bnpparibas.com","more_info":"the following user(s) are already in processing state, please try again after one minute"}],"status_code":409}
```


### Expected Behavior

It should work as the user does not exist on the account

### Actual Behavior

Logs in debug mode indicates a first execution of the api with 202 return code then a second execution with 409 return code 

│ Error: Request failed with status code: 409, ServerErrorResponse: {"trace":"10778924-a33c-4684-b720-04d63a9867e2","errors":[{"code":"Conflict","message":"Invite request blocked : gaston.lagaffe@bnpparibas.com","more_info":"the following user(s) are already in processing state, please try again after one minute"}],"status_code":409}
│ 
│   with ibm_iam_user_invite.access_group_user_management,
│   on main.tf line 183, in resource "ibm_iam_user_invite" "access_group_user_management":
│  183: resource "ibm_iam_user_invite" "access_group_user_management" {
│ 
│ ---
│ id: terraform-fa2a7c87
│ summary: 'Request failed with status code: 409, ServerErrorResponse:
│ {"trace":"10778924-a33c-4684-b720-04d63a9867e2","errors":[{"code":"Conflict","message":"Invite
│   request blocked : gaston.lagaffe@bnpparibas.com","more_info":"the following user(s)
│   are already in processing state, please try again after one minute"}],"status_code":409}'
│ severity: error
│ resource: ibm_iam_user_invite
│ operation: create
│ component:
│   name: github.com/IBM-Cloud/terraform-provider-ibm
│   version: 1.87.3
│ ---

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply -auto-approve`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

* #0000

https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/iam_user_invite#inviting-batch-of-users

--- a/ibm/acctest/acctest.go
+++ b/ibm/acctest/acctest.go
@@ -67,6 +67,7 @@ var (
 	FloatingIpID                    string
 	HpcsInstanceID                  string
 	HpcsInstanceName                string
+	IAMAccessGroupId                string
 	IAMAccountId                    string
 	IAMServiceId                    string
 	IAMTrustedProfileID             string
@@ -536,6 +537,11 @@ func init() {
 	IAMUser = os.Getenv("IBM_IAMUSER")
 	if IAMUser == "" {
 		fmt.Println("[WARN] Set the environment variable IBM_IAMUSER for testing ibm_iam_user_policy resource Some tests for that resource will fail if this is not set correctly")
+	}
+
+	IAMAccessGroupId = os.Getenv("IBM_IAM_ACCESS_GROUP_ID")
+	if IAMAccessGroupId == "" {
+		fmt.Println("[WARN] Set the environment variable IBM_IAM_ACCESS_GROUP_ID for testing ibm_iam_user_invite resource Some tests for that resource will fail if this is not set correctly")
 	}
 
 	IAMAccountId = os.Getenv("IBM_IAMACCOUNTID")

--- a/ibm/service/iampolicy/resource_ibm_iam_user_invite.go
+++ b/ibm/service/iampolicy/resource_ibm_iam_user_invite.go
@@ -682,7 +682,7 @@ func resourceIBMIAMUpdateUserProfile(d *schema.ResourceData, meta interface{}) e
 	}
 	Client := userManagement.UserInvite()
 
-	if d.HasChange("users") {
+	if d.HasChange("users") && !d.IsNewResource() {
 		//var removedUsers, addedUsers []string
 		accountID, err := getAccountID(d, meta)
 		if err != nil {

--- a/ibm/service/iampolicy/resource_ibm_iam_user_invite_test.go
+++ b/ibm/service/iampolicy/resource_ibm_iam_user_invite_test.go
@@ -1,0 +1,71 @@
+// Copyright IBM Corp. 2017, 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package iampolicy_test
+
+import (
+	"fmt"
+	"testing"
+
+	acc "github.com/IBM-Cloud/terraform-provider-ibm/ibm/acctest"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccIBMIAMUserInvite_Basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acc.TestAccPreCheck(t) },
+		Providers:    acc.TestAccProviders,
+		CheckDestroy: testAccCheckIBMIAMUserInviteDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckIBMIAMUserInviteBasic(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckIBMIAMUserInviteExists("ibm_iam_user_invite.invite_user"),
+					resource.TestCheckResourceAttr("ibm_iam_user_invite.invite_user", "users.#", "1"),
+					resource.TestCheckResourceAttr("ibm_iam_user_invite.invite_user", "access_groups.#", "1"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckIBMIAMUserInviteDestroy(s *terraform.State) error {
+	// User invites are transient - once processed, they don't persist as resources
+	// The invited users become part of the account, but the invite itself is gone
+	// So there's nothing to check for destroy
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "ibm_iam_user_invite" {
+			continue
+		}
+		// Just verify the resource was in state
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set for user invite resource")
+		}
+	}
+	return nil
+}
+
+func testAccCheckIBMIAMUserInviteExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No user invite ID is set")
+		}
+		return nil
+	}
+}
+
+func testAccCheckIBMIAMUserInviteBasic() string {
+	return fmt.Sprintf(`
+		resource "ibm_iam_user_invite" "invite_user" {
+			users = ["%s"]
+			access_groups = ["%s"]
+		}
+	`, acc.IAMUser, acc.IAMAccessGroupId)
+}

